framework/src/main/java/org/checkerframework/framework/util/JavaExpressionParseUtil.java

```diff
@@ -951,19 +967,28 @@ public JavaExpression visit(BinaryExpr expr, Void aVoid) {
       TypeMirror leftType = leftJe.getType();
       TypeMirror rightType = rightJe.getType();
       TypeMirror type;
+      Supplier<ParseRuntimeException> exception =
```

Comment:
This is a clever way to save two lines in the code, at the cost of creating a lambda at run time that might never be used.
Could you please make this change a separate pull request, so that there isn't a pull request that contains both this refactoring and a bug fix?